app/src/main/java/com/duckduckgo/app/fire/LottieAnimationLoader.kt

```diff
@@ -21,21 +21,25 @@ import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.LifecycleObserver
 import androidx.lifecycle.OnLifecycleEvent
 import com.airbnb.lottie.LottieCompositionFactory
-import com.duckduckgo.app.browser.R
-import com.duckduckgo.app.statistics.VariantManager
+import com.duckduckgo.app.settings.db.SettingsDataStore
 import javax.inject.Inject
 
-class FireAnimationLoader @Inject constructor(
-    private val variantManager: VariantManager,
-    private val context: Context
-) : LifecycleObserver {
+interface AnimationLoader : LifecycleObserver {
```

Comment:
I think this file is now an implementation detail of settings, I would move it into the settings package.
Also the naming is now confusing to me, it very specifically preloads file animations, however the interface and its impl names are very generic, as if it could be used for any animation.
I think the prev FireAnimationLoader naming was better

Reply:
Very good points. I would move it to the settings package.

And about naming, using `FireAnimationLoader` sounds good now that the concrete class is `LottieAnimationLoader`.  TBH, all this code is pretty tied to Lottie because we want to precache the json file to improve performance, and I would have avoided an interface here. But the interface is because we can't `mock` final classes in AndroidTests...so it was tricky to find some generic names for something very specific. But I think your proposal is good.

Reply:
Same as @aitorvs said, it feels like `FireAnimationLoader` was a better name. I feel that if I see `AnimationLoader` in the codebase I might be tempted to use it for other animation related stuff.

How about changing the interface to `FireAnimationLoader` and the implementation to `LottieFireAnimationLoader`?